src/DataTypes/array.tsx

```diff
@@ -0,0 +1,161 @@
+import type { JSONSchema7 as JSONSchema } from 'json-schema';
+import {
+	FULL_TEXT_SLUG,
+	createModelFilter,
+	type FormData,
+} from '../components/Filters/SchemaSieve';
+import { CreateFilter, getDataTypeSchema } from './utils';
+import { getDataModel } from '.';
+import { isJSONSchema, getRefSchema } from '../AutoUI/models/helpers';
+
+export const operators = () => ({
+	contains: 'contains',
+	not_contains: 'does not contain',
+});
+
+export type OperatorSlug =
+	| keyof ReturnType<typeof operators>
+	| typeof FULL_TEXT_SLUG;
+
+const isSchemaWithPrimitiveItems = (
+	schema: JSONSchema,
+): schema is JSONSchema & { items: JSONSchema } =>
+	!!schema?.items &&
+	typeof schema.items !== 'boolean' &&
+	'type' in schema.items;
+
+const isArrayOfObjectSchema = (schema: JSONSchema): boolean => {
+	return (
+		!!schema?.items &&
+		typeof schema.items !== 'boolean' &&
+		'properties' in schema.items &&
+		!!schema.items.properties
+	);
+};
+
+const buildFilterForPropertySchema = (
+	field: string,
+	operator: OperatorSlug,
+	value: string,
+	schema: JSONSchema,
+): JSONSchema | {} => {
+	const filter = getFilter(field, schema, value, operator);
+	if (!Object.keys(filter).length) {
+		return {};
+	}
+
+	return {
+		properties: { [field]: { type: 'array', ...filter } },
+		required: [field],
+	};
+};
+const wrapFilter = (filter: JSONSchema): JSONSchema => {
+	if (typeof filter.not === 'object') {
+		return { not: { contains: filter.not } };
+	}
+	return { minItems: 1, contains: filter };
+};
+
+const determineOperator = (
+	operator: OperatorSlug,
+	schema: JSONSchema,
+): OperatorSlug | 'is' | 'is_not' => {
+	if (operator === 'not_contains') {
+		return 'is_not';
+	}
+	if (
+		operator === 'contains' ||
+		(schema.type === 'string' && operator === FULL_TEXT_SLUG)
+	) {
+		return 'is';
+	}
+	return operator;
```

Comment:
convert to switch?

Reply:
This was my initial thought as well, but I would prefer to keep all DataTypes having the same style. Since in some DataType cases we need to check more then 1 property I needed to use `if` statements for these cases. I think that keeping if for all would make DataTypes easier to read